### PR TITLE
[mds-policy-author][mds-db] Adding bulk read policy and geography metadata endpoints

### PR DESCRIPTION
**Pull request by janedotx**
## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [x] Policy Author

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -278,6 +278,7 @@
       "request": "launch",
       "name": "Tests: mds-db",
       "program": "${workspaceFolder}/node_modules/mocha/bin/_mocha",
+      "runtimeArgs": ["--harmony"],
```
Comment (avatarneil) [inline .vscode/launch.json]: I think you added this to try and get `async/await` working nicely in the debugger, but IIRC it didn't work? Might as well remove

File: packages/mds-db/index.ts [hunk 2]
```diff
@@ -1068,20 +1069,39 @@ async function readGeographies(params?: { geography_id?: UUID; get_read_only?: b
...
+      return row
+    })
   } catch (err) {
     await log.error('readGeographies', err)
     throw err
   }
 }
 
+async function readBulkGeographyMetadata(params?: {
+  geography_id?: UUID
```
[the remark this answers is not in the document]
Comment (janedotx) [reply packages/mds-db/index.ts]: per our in-person discussion yesterday, i will remove this and add a `readSingleGeography` function to `mds-db`

File: packages/mds-db/index.ts [hunk 3]
```diff
@@ -1213,6 +1240,44 @@ async function readPolicies(params?: {
   return res.rows.map(row => row.policy_json)
 }
 
+async function readBulkPolicyMetadata(params?: {
+  policy_id?: UUID
```
[the remark this answers is not in the document]
Comment (janedotx) [reply packages/mds-db/index.ts]: They're used in the call to `readPolicies` though.

File: packages/mds-db/index.ts [hunk 4]
```diff
@@ -1134,32 +1154,39 @@ async function deleteGeography(geography_id: UUID) {
...
+  })
   const {
     rows: [recorded_metadata]
   }: { rows: Recorded<Geography>[] } = await client.query(sql, values)
-  return { ...metadata, ...recorded_metadata }
+  return { ...geography_metadata, ...recorded_metadata }
 }
 
-async function readGeographyMetadata(geography_id: UUID): Promise<GeographyMetadata> {
+async function readSingleGeographyMetadata(geography_id: UUID): Promise<GeographyMetadata> {
```
[the remark this answers is not in the document]
Comment (janedotx) [reply packages/mds-db/index.ts]: Well, I don't think it makes a lot of sense to have additional params if you know you're expecting a single `Geography` that matches the UUID that you sent. For bulk geography reads, I think it makes sense to be able to sort the published/read-only ones from the unpublished ones.

Now that you've pointed it out, it is true we don't have pagination params for bulk reads. Do you think I should add those?

File: .vscode/launch.json (hunk 1 above)
Comment (janedotx) [reply .vscode/launch.json]: will do

File: packages/mds-db/index.ts [hunk 5]
```diff
@@ -1068,20 +1069,39 @@ async function readGeographies(params?: { geography_id?: UUID; get_read_only?: b
...
-        throw new NotFoundError(`geography ${params.geography_id}`)
-      } else {
-        throw new NotFoundError(`geographies not found`)
-      }
-    }
-    return rows
+
+    return rows.map(row => {
+      /* eslint-disable-next-line no-param-reassign */
+      delete row.id
```
[the remark this answers is not in the document]
Comment (janedotx) [reply packages/mds-db/index.ts]: Acknowledged.

**Comment by janedotx** [comment]
@avatarneil addressed feedback. Also the PUTs verbs do creates AND updates now for the policy and geography metadata. Not having those earlier was a derp on my part

**Comment by avatarneil** [comment]
@janedotx Looks like the linter is detecting problems on CI/CD

**Comment by avatarneil** [comment]
@twelch would you mind reviewing this and verifying (at least by checking out the test cases) that it behaves how you expect before we merge?

**Comment by janedotx** [comment]
> @twelch would you mind reviewing this and verifying (at least by checking out the test cases) that it behaves how you expect before we merge?

Tim and Owen have told me they'd prefer to hit it on sandbox.

**Pull request merged**